src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Simplification/Simplifiers/CastSimplifier.cs

```diff
@@ -347,6 +347,9 @@ public static bool IsUnnecessaryAsCast(BinaryExpressionSyntax cast, SemanticMode
             if (castType.Kind == SymbolKind.DynamicType)
                 return true;
 
+            if (EnumCastMustBePreserved(castNode, castType, castedExpressionType))
+                return false;
```

Comment:
this is flipped.  but it turns out htis code doesn't even matter.  removing later.